src/launchpad/utils/metadata_extractor.py

```diff
@@ -0,0 +1,90 @@
+"""Utility for extracting metadata from .sentry-cli-metadata.txt files in artifacts."""
+
+import zipfile
+
+from pathlib import Path
+from typing import Dict, Optional
+
+from launchpad.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
+METADATA_FILENAME = ".sentry-cli-metadata.txt"
+
+
+class ToolingMetadata:
+    """Container for tooling version metadata extracted from artifacts."""
+
+    def __init__(
+        self,
+        cli_version: Optional[str] = None,
+        fastlane_plugin_version: Optional[str] = None,
+        gradle_plugin_version: Optional[str] = None,
+    ):
+        self.cli_version = cli_version
+        self.fastlane_plugin_version = fastlane_plugin_version
+        self.gradle_plugin_version = gradle_plugin_version
+
+    def __repr__(self) -> str:
+        return f"ToolingMetadata(cli_version={self.cli_version}, fastlane_plugin_version={self.fastlane_plugin_version}, gradle_plugin_version={self.gradle_plugin_version})"
+
+
+def extract_metadata_from_zip(zip_path: Path) -> ToolingMetadata:
+    """Extract tooling metadata from a .sentry-cli-metadata.txt file in the root of a zip.
+
+    Args:
+        zip_path: Path to the zip file to search
+
+    Returns:
+        ToolingMetadata object with extracted version information
+    """
+    try:
+        with zipfile.ZipFile(zip_path, "r") as zf:
+            # Only look for .sentry-cli-metadata.txt in the root of the zip
+            if METADATA_FILENAME not in zf.namelist():
+                logger.debug(f"No {METADATA_FILENAME} found in root of {zip_path}")
+                return ToolingMetadata()
+
+            logger.debug(f"Found metadata file: {METADATA_FILENAME}")
+
+            with zf.open(METADATA_FILENAME) as f:
+                content = f.read().decode("utf-8")
+                return _parse_metadata_content(content)
+
+    except Exception as e:
+        logger.warning(f"Failed to extract metadata from {zip_path}: {e}")
+        return ToolingMetadata()
+
+
+def _parse_metadata_content(content: str) -> ToolingMetadata:
+    """Parse the content of .sentry-cli-metadata.txt file.
+
+    Expected format:
+        sentry-cli-version: 2.58.2
+        sentry-fastlane-plugin: 1.2.3
+        sentry-gradle-plugin: 4.12.0
+
+    Args:
```

Comment:
IMO these arg/return comments aren't useful, the format is useful IMO so you can leave that in